gunrock/app/SimpleTemplate/SimpleTemplate_app.cu

```diff
@@ -0,0 +1,341 @@
+// ----------------------------------------------------------------------------
+// Gunrock -- Fast and Efficient GPU Graph Library
+// ----------------------------------------------------------------------------
+// This source code is distributed under the terms of LICENSE.TXT
+// in the root directory of this source distribution.
+// ----------------------------------------------------------------------------
+
+/**
+ * @file SimpleTemplate_app.cu
+ *
+ * @brief Simple Gunrock Application
+ */
+
+#include <gunrock/gunrock.h>
+#include <gunrock/util/test_utils.cuh>
+#include <gunrock/graphio/graphio.cuh>
+#include <gunrock/app/app_base.cuh>
+#include <gunrock/app/test_base.cuh>
+
+// <todo> change includes
+#include <gunrock/app/SimpleTemplate/SimpleTemplate_enactor.cuh>
+#include <gunrock/app/SimpleTemplate/SimpleTemplate_test.cuh>
+// </todo>
+
+namespace gunrock {
+namespace app {
+// <todo> change namespace
+namespace SimpleTemplate {
+// </todo>
+
+
+cudaError_t UseParameters(util::Parameters &parameters)
+{
+    cudaError_t retval = cudaSuccess;
+    GUARD_CU(UseParameters_app(parameters));
+    GUARD_CU(UseParameters_problem(parameters));
+    GUARD_CU(UseParameters_enactor(parameters));
+
+    // <todo> add app specific parameters, eg:
+    // GUARD_CU(parameters.Use<std::string>(
+    //    "src",
+    //    util::REQUIRED_ARGUMENT | util::MULTI_VALUE | util::OPTIONAL_PARAMETER,
+    //    "0",
+    //    "<Vertex-ID|random|largestdegree> The source vertices\n"
+    //    "\tIf random, randomly select non-zero degree vertices;\n"
+    //    "\tIf largestdegree, select vertices with largest degrees",
+    //    __FILE__, __LINE__));
+    // </todo>
+
+    return retval;
+}
+
+/**
+ * @brief Run SimpleTemplate tests
+ * @tparam     GraphT        Type of the graph
+ * @tparam     ValueT        Type of the distances
+ * @param[in]  parameters    Excution parameters
+ * @param[in]  graph         Input graph
+...
+ * @param[in]  target        where to perform the app 
+ * \return cudaError_t error message(s), if any
+ */
```

Comment:
@jowens has noted before that we need to be more detailed with commenting the parameters.